src/domain/action-item.ts

```diff
@@ -126,8 +126,11 @@ export class VstsRelease extends Build {
     this.name = releaseInfo.releaseDefinition.name;
     this.created = moment(releaseInfo.createdOn).valueOf();
     this.url = releaseInfo._links.web.href;
-    const inProgressEnvs = releaseInfo.environments.filter(env => env.status === 'inProgress');
-    this.building = inProgressEnvs.length > 0;
+    const isInProgress = releaseInfo.environments.some(env => env.status === 'inProgress');
+    const requiresApproval = releaseInfo.environments
+      .some(env => env.preDeployApprovals && env.preDeployApprovals
+        .some(approval => approval.status === 'pending'));
```

Comment:
this would be a bit easier to read as follows:

```
const containsPendingApproval = (env) => {
  return env.preDeployApprovals && env.preDeployApprovals.some(approval => approval.status === 'pending');
}
const requiresApproval = releaseInfo.environments.some(containsPendingApproval);

Reply:
hmm, i get it... but not sure it makes it easier to read, lol, it took me a minute to figure out that `.some(containsPendingApproval)` was the function reference, if this is the _way_ then i'm on board